lib/notifier/email.js

```diff
@@ -0,0 +1,102 @@
+const _ = require('lodash/fp')
+const utils = require("./utils")
+
+const email = require('../email')
+
+const {
+  PING,
+  STALE,
+  LOW_CRYPTO_BALANCE,
+  HIGH_CRYPTO_BALANCE,
+  CASH_BOX_FULL,
+  LOW_CASH_OUT
+} = require('./codes')
+
+function alertSubject(alertRec, config) {
+  let alerts = []
+
+  if (config.balance) {
+    alerts = _.concat(alerts, alertRec.general)
+  }
+
+  _.keys(alertRec.devices).forEach(function (device) {
+    if (config.balance) {
+      alerts = _.concat(alerts, alertRec.devices[device].balanceAlerts)
+    }
+
+    if (config.errors) {
+      alerts = _.concat(alerts, alertRec.devices[device].deviceAlerts)
+    }
+  })
+
+  if (alerts.length === 0) return null
+
+  const alertTypes = _.map(codeDisplay, _.uniq(_.map('code', alerts))).sort()
+  return '[Lamassu] Errors reported: ' + alertTypes.join(', ')
+}
+
+function printEmailAlerts(alertRec, config) {
+  let body = 'Errors were reported by your Lamassu Machines.\n'
+
+  if (config.balance && alertRec.general.length !== 0) {
+    body += '\nGeneral errors:\n'
+    body += emailAlerts(alertRec.general) + '\n'
+  }
+
+  _.keys(alertRec.devices).forEach(function (device) {
+    const deviceName = alertRec.deviceNames[device]
+    body += '\nErrors for ' + deviceName + ':\n'
+
+    let alerts = []
+    if (config.balance) {
+      alerts = _.concat(alerts, alertRec.devices[device].balanceAlerts)
+    }
+
+    if (config.errors) {
+      alerts = _.concat(alerts, alertRec.devices[device].deviceAlerts)
+    }
```

Comment:
Same issues as above, use `_.forEach()`. Also, consider a Pull Up Method refactoring somewhere on this section of code, as the functionality seems to be similar to the snippet a bit above

```suggestion
  _.forEach(_.keys(alertRec.devices), (device) => {
    const deviceName = alertRec.deviceNames[device]
    body += '\nErrors for ' + deviceName + ':\n'
    let alerts = []
	
    if (config.balance)
      alerts = _.concat(alerts, alertRec.devices[device].balanceAlerts)

    if (config.errors)
      alerts = _.concat(alerts, alertRec.devices[device].deviceAlerts)
  })
```